docs/en/userguide/advanced_setup.md

```diff
@@ -0,0 +1,277 @@
+    {
+        // disallow elemental data inheritance in the case that published localised page instance already exists
+        if ($field == 'ElementalAreaID' && $this->owner->isPublishedInLocale()) {
+            $query = '"' . $table . '_Localised_' . $locale->getLocale() . '"."' . $field . '"';
+        }
+    }
+}
+```
+
+
+```
+class BlockPage extends Page
+{
+    .
+    .
+    .
+
+    /**
+     * @var array
+     */
+    private static $extensions = [
+        ElementalPageExtension::class,
+        BlockPageFluentExtension::class,
+    ];
+```
+
+#### Benefits of indirect localisation
+
+* different localisation of pages can have completely different set of blocks which allows greater flexibility
+* localisation is only on page level, so any functionality on block level does not need to care about localisation
+* this is especially useful when writing unit tests as is significantly easier to set up tests without localisation
+
+#### Downsides of indirect localisation
+
+* the blocks are unaware of their locales which makes bottom up relation lookup slower
+* this can be remedied by some extra data stored in blocks (see notes below)
+
+### Block performance increase
+
+If only one page can own a block, we can store page reference directly on the block.
+This can be done when block is created (`onBeforeWrite`). This is helps significantly when traversing relation from block up to a page.
+Note that there may be a lot of objects sitting between block and a page.
+For example `Content block -> Elemental area -> Layout block -> Elemental area -> Page`.
+
+Additional information can be stored like page locale (relevant for `Fluent` module) to specify the target data even further.
+
+### Unit tests
+
+Writing unit tests for `Elemental` with `Fluent` can be rather tricky to figure out.
+Here are some guidelines to make that easier.
+
+#### Make sure your fixture has some locales setup
+
+It's important to include some locales because otherwise your test might be testing a very different situation.
+Example `Locale` setup in a fixture:
+
+```
+TractorCow\Fluent\Model\Locale:
+  nz:
+    Locale: en_NZ
+    Title: 'English (New Zealand)'
+    URLSegment: newzealand
+  us:
+    Locale: en_US
+    Title: 'English (US)'
+    URLSegment: usa
+    Fallbacks:
+      - =>TractorCow\Fluent\Model\Locale.nz
+```
+
+#### Localised fixture data (automatic, single locale)
+
+In the case your fixture needs to contain data for only a single locale you can specify your desired locale in your unit test like this:
+
+```
+protected function setUp()
+{
+    // Set locale for fixture creation
+    FluentState::singleton()->withState(function (FluentState $state): void {
+        $state->setLocale('en_NZ');
+        parent::setUp();
+    });
+}
+
+```
+
+This will localise all your data so you don't need to worry about that in your fixtures. Following fixture will produce a page localised in `en_NZ`:
```

Comment:
```suggestion
This will localise all your data so you don't need to worry about that in your fixtures. The following fixture will produce a page localised in `en_NZ`:
```